src/transport/raw/PeerAddress.h

```diff
@@ -133,18 +136,28 @@ class PeerAddress
     {
         char ip_addr[kInetMaxAddrLen];
 
+        char interface[Inet::InterfaceId::kMaxIfNameLength + 2] = {'%', 0};
+        if (mInterface.IsPresent()) {
+            if (mInterface.GetInterfaceName(interface + 1, sizeof(interface) - 1) != CHIP_NO_ERROR)
+            {
+                ::strncpy(interface, "%(err)", sizeof(interface));
+            }
+        } else {
+            ::strncpy(interface, "%(nul)", sizeof(interface));
+        }
+
         switch (mTransportType)
         {
         case Type::kUndefined:
             snprintf(buf, bufSize, "UNDEFINED");
             break;
         case Type::kUdp:
             mIPAddress.ToString(ip_addr);
-            snprintf(buf, bufSize, "UDP:%s:%d", ip_addr, mPort);
+            snprintf(buf, bufSize, "UDP:[%s]:%d%s", ip_addr, mPort, interface);
```

Comment:
I'm trying to advocate for an output format that can be parsed by some standard tools because this saves us from having to invent our own grammar, and this also allows us to use log output directly in some cases.

These things are counter to that:

* use of square brackets around an IPv4 addresses
* interface printed after port, not after IP address
* printing %(nul) as interface when none is known or relevant

As mentioned previously, this is a command that I can actually enter into a terminal, and there are others that also work like this:

```
socat - TCP6:[fe80::6fe:bcc9:93a:5b35%enp0s3]:700
```

This comment suggest how approximately one might get to this output style:

https://github.com/project-chip/connectedhomeip/pull/11176#discussion_r740318248

Reply:
Thanks, I'll use this.